### DISCUSSION TITLE
Support for JSON mode

### DISCUSSION THREAD

**Discussion by DevShaded** [openai-php/client#350] [General]
In the Chat Completion there is a mode you can enable in Text generation called [JSON mode](https://platform.openai.com/docs/guides/text-generation/json-mode). When will this be added to the client?

**Comment by hschimpf** [comment]
You can pass the `json mode` in the parameters of the request.

```diff
$client::chat()->create([
    'model' => 'gpt-4',
+   'response_format' => 'json_object',
    // ...
]);
```

**Comment by ericthornton88** [comment]
Putting `json_object` as a string didn't work for me. Here is how I had to format it in order for it to work
```php
'model' => 'gpt-4o',
'response_format' => [
     'type' => 'json_object'
],
```

**Reply by hschimpf** [reply]
You should also instruct the model to respond with a JSON[^1].

[^1]: [JSON Mode - OpenAI API](https://platform.openai.com/docs/guides/json-mode/json-mode)
